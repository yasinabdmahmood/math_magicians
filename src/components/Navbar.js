import { Link } from 'react-router-dom';

const Navbar = () => (
  <div className="nav-container">
    <h1>Math Magician</h1>
    <ul className="navbar">
      <li data-testid="li"><Link to="/">Home</Link></li>
      <li data-testid="li"><Link to="/Calculator">Calculator</Link></li>
      <li data-testid="li"><Link to="/Quote">Quote</Link></li>
    </ul>
  </div>
);
export default Navbar;
