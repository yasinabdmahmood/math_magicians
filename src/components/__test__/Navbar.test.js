import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

const MockTodoFooter = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <MockTodoFooter />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render three li element', () => {
  render(
    <MockTodoFooter />,
  );
  const listItems = screen.getAllByTestId('li');

  expect(listItems.length).toBe(3);
});
