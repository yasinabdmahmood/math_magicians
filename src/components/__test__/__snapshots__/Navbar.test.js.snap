// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<div
  className="nav-container"
>
  <h1>
    Math Magician
  </h1>
  <ul
    className="navbar"
  >
    <li
      data-testid="li"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      data-testid="li"
    >
      <a
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      data-testid="li"
    >
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</div>
`;
